### PR TITLE
fix(prometheus): regexp handling should comply with PromQL

### LINKED ISSUE #19831: remote read: regular expression handling not consistent with Prometheus
Prometheus handles regular expression in a special, sort of counterintuitive way that it adds  begin/end boundaries automatically for users. But influxdb does not follow this way to process remote read RPC. 

Example PromQL: 
`go_gc_duration_seconds{quantile=~"0"}`

native prometheus shows:

```
go_gc_duration_seconds{instance="localhost:9090",job="prometheus",quantile="0"} | 0.000080217
```

influxdb as remote read storage:

```
go_gc_duration_seconds{instance="localhost:9090",job="prometheus",quantile="0"} | 0.000080217
go_gc_duration_seconds{instance="localhost:9090",job="prometheus",quantile="0.25"} | 0.000084891
go_gc_duration_seconds{instance="localhost:9090",job="prometheus",quantile="0.5"} | 0.000098008
go_gc_duration_seconds{instance="localhost:9090",job="prometheus",quantile="0.75"} | 0.000126765
```

### PR DESCRIPTION
**Pull request by foobar**
Closes #19831

### CODE REVIEW

**Comment by foobar** [comment]
@davidby-influx could you take a look?

File: prometheus/converters.go [hunk 1]
```diff
@@ -240,7 +240,9 @@ func nodeFromMatcher(m *remote.LabelMatcher) (*datatypes.Node, error) {
 		right = &datatypes.Node{
 			NodeType: datatypes.NodeTypeLiteral,
 			Value: &datatypes.Node_RegexValue{
-				RegexValue: m.Value,
+				// To comply with PromQL, see
+				// https://github.com/prometheus/prometheus/blob/daf382e4a9f5ca380b2b662c8e60755a56675f14/pkg/labels/regexp.go#L30
+				RegexValue: "^(?:" + m.Value + ")$",
```
Comment (davidby-influx) [inline prometheus/converters.go:245]: Thanks for the PR.  I'm new to this code, so please excuse any misunderstandings from my ignorance.

I see that Prometheus had this construct `"^(?:" + v + ")$"` in the code before the addition of the FastRegexMatcher:

https://github.com/prometheus/prometheus/commit/cef4dd6fff06198c61007211be460fb83cd708c1#diff-7593b5b403ae21dae6d08a8bf75092a4792a11dbfd88dd15ecd33b15ea2611c4L62

The change with anchoring happened in this commit: 

https://github.com/prometheus/prometheus/commit/33c1171b9c50296ce61b86c0c4b2dfc19aea6545#diff-7593b5b403ae21dae6d08a8bf75092a4792a11dbfd88dd15ecd33b15ea2611c4

Did something break that prompted your PR?

Thanks for helping me to understand.

Comment (foobar) [reply prometheus/converters.go:245]: afaics,  FastRegexMatcher also does
```
	re, err := regexp.Compile("^(?:" + v + ")$")
```

In practice, PromQL handles regex with anchor explicitly, as the example in #19831

**Review by danxmoran — APPROVED** [review]
This makes sense to me based on the linked discussions

**Review by davidby-influx — APPROVED** [review]
Seems good

**Pull request merged**